### DISCUSSION TITLE
Config file

### DISCUSSION THREAD

**Discussion by th-art001** [lllyasviel/Fooocus#940] [Q&A]
Hi! Why my config file refresh with every launch of Fooocus (2.1.803)? I'd like to decrease a count of dimensions in advance settings, but  can't, because after editing and saving config file, it back for initial condition.  ![Screenshot_16](https://github.com/lllyasviel/Fooocus/assets/59409653/e6927dc1-6609-4098-ac4e-9a6b2caea313)

**Comment by lllyasviel** [comment]
* use \\\\ not \\ after editing paths
* make sure that there is no `,` before the last `}`
* make sure the paths do exist

**Reply by th-art001** [reply]
You misunderstand me. I don't edit the paths. I just want to change "aspect ratios" section. 
Just leave several options, something like that:
"available_aspect_ratios"
        "720 * 960",
        "768 * 768",
        "960 * 720",
    "default_aspect_ratio": "768*768"
    
   But my config.txt back to default condition, like I'm running Fooocus for the first time. It's not saving. (using Notepad++)
  
    
![Screenshot_3](https://github.com/lllyasviel/Fooocus/assets/59409653/a75a1011-ad42-4b7c-85a2-dfb0b1dd20e2)

**Reply by lllyasviel** [reply]
see also tutorials in `config_modification_tutorial.txt`
if you believe everything is correct, post modified config

**Comment by aolko** [comment]
So, yeah, fooocus reverts the config on **ANY syntax error**

**Comment by thiagojramos** [comment]
The same issue with me. Whenever I edit the "config.txt" and start any of the ".bat" files, it resets the "config.txt" to the default values. This is a bigger problem for me because, in addition to moving all the models to another folder elsewhere, I separate the models (e.g., loras or checkpoints) into folders (e.g., SD15, SDXL). And besides resetting the config.txt with each run, it downloads models (which I often already have in the folders) to the root folder of the model type. 😣😔

**Reply by thiagojramos** [reply]
~~I tried to prevent config.txt from being altered by setting it to "read-only," but when I start the .bat file (even though it's customized, it normally invokes entry_with_update.py with arguments), it gives a permission denied error. Well, it was an attempt to avoid being altered by default.~~

<details><summary>Console log</summary>

```powershell
Choose an option:
[1] Default
[2] Anime
[3] Realistic

Enter your choice (1-3): 2
Already up-to-date
Update succeeded.
[System ARGV] ['C:\\Fooocus\\Fooocus\\entry_with_update.py', '--preset', 'anime', '--normalvram', '--language', 'pt_BR']
Traceback (most recent call last):
  File "C:\Fooocus\Fooocus\entry_with_update.py", line 46, in <module>
    from launch import *
  File "C:\Fooocus\Fooocus\launch.py", line 22, in <module>
    from modules.config import path_checkpoints, path_loras, path_vae_approx, path_fooocus_expansion, \
  File "C:\Fooocus\Fooocus\modules\config.py", line 283, in <module>
    with open(config_path, "w", encoding="utf-8") as json_file:
PermissionError: [Errno 13] Permission denied: 'C:\\Fooocus\\Fooocus\\config.txt'
Press any key to continue. . .
``` 

</details>

~~My custom config.txt:~~

~~[config.txt](https://github.com/lllyasviel/Fooocus/files/13370763/config.txt)~~

**Reply by thiagojramos** [reply]
After the last update, it seems that the config.txt is no longer being reset to the default values. However, one thing I noticed is that if the models are in a subfolder, as in my case, you have to include their extension. For example, "modelname.safetensors"
